Mathlib/Topology/ContinuousFunction/CompactlySupported.lean

```diff
@@ -0,0 +1,709 @@
+/-
+Copyright (c) 2024 Yoh Tanimoto. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yoh Tanimoto
+-/
+import Mathlib.Topology.ContinuousFunction.Bounded
+import Mathlib.Topology.ContinuousFunction.CocompactMap
+import Mathlib.Topology.ContinuousFunction.Compact
+import Mathlib.Topology.ContinuousFunction.ZeroAtInfty
+
+/-!
+# Compactly supported continuous functions
+
+The type of compactly supported continuous functions. When the domain is compact,
+`C(α, β) ≃ C_c(α, β)` via the identity map. When the codomain is a metric space, every continuous
+compactly supported map is a bounded continuous function. When the domain is a locally
+compact space, this type has nice properties.
+
+## TODO
+
+* Create more intances of algebraic structures (e.g., `NonUnitalSemiring`) once the necessary
+  type classes (e.g., `TopologicalRing`) are sufficiently generalized.
+-/
+
+
+universe u v w
+
+variable {F : Type*} {α : Type u} {β : Type v} {γ : Type w} [TopologicalSpace α]
+
+open BoundedContinuousFunction Topology Bornology
+
+open Filter Metric
+
+/-- `C_c(α, β)` is the type of continuous functions `α → β` with compact support from a topological
+space to a topological space with a zero element.
+
+When possible, instead of parametrizing results over `(f : C_c(α, β))`,
+you should parametrize over `(F : Type*) [CompactlySupportedContinuousMapClass F α β] (f : F)`.
+
+When you extend this structure, make sure to extend `CompactlySupportedContinuousMapClass`. -/
+structure CompactlySupportedContinuousMap (α : Type u) (β : Type v) [TopologicalSpace α] [Zero β]
+    [TopologicalSpace β] extends ContinuousMap α β : Type max u v where
+  /-- The function has compact support . -/
+  has_compact_support' : HasCompactSupport toFun
```

Comment:
Try to match the symbols. No underscores in HasCompactSupport, so no reason why you should introduce one here.
```suggestion
  hasCompactSupport' : HasCompactSupport toFun
```